test/integration/extensionRequests.test.js

```diff
@@ -0,0 +1,631 @@
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Extension Request created successfully!");
+          expect(res.body.extensionRequest).to.be.a("object");
+          expect(res.body.extensionRequest.assignee).to.equal(appOwner.id);
+          expect(res.body.extensionRequest.status).to.equal(ETA_EXTENSION_REQUEST_STATUS.PENDING);
+          return done();
+        });
+    });
+    it("Should return fail response if someone try to create a extension request for someone else and is not a super user", function (done) {
+      chai
+        .request(app)
+        .post("/extensionRequests")
+        .set("cookie", `${cookieName}=${appOwnerjwt}`)
+        .send({
+          taskId: taskId2,
+          title: "change ETA",
+          assignee: user.id,
+          oldEndsOn: 1234,
+          newEndsOn: 1235,
+          reason: "family event",
+          status: "PENDING",
+        })
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(403);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Only Super User can create an extension request for this task.");
+          return done();
+        });
+    });
+    it("Should return fail response if task with the taskId doesn't exists", function (done) {
+      chai
+        .request(app)
+        .post("/extensionRequests")
+        .set("cookie", `${cookieName}=${appOwnerjwt}`)
+        .send({
+          taskId: "12345678",
+          title: "change ETA",
+          assignee: appOwner.id,
+          oldEndsOn: 1234,
+          newEndsOn: 1235,
+          reason: "family event",
+          status: "PENDING",
+        })
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(400);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Task with taskId doesn't exist");
+          return done();
+        });
+    });
+    it("Should return fail response if task belongs to someone else", function (done) {
+      chai
+        .request(app)
+        .post("/extensionRequests")
+        .set("cookie", `${cookieName}=${appOwnerjwt}`)
+        .send({
+          taskId: taskId2,
+          title: "change ETA",
+          assignee: appOwner.id,
+          oldEndsOn: 1234,
+          newEndsOn: 1235,
+          reason: "family event",
+          status: "PENDING",
+        })
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(400);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("This task is assigned to some different user");
+          return done();
+        });
+    });
+    it("Should return fail response if the new ETA is invalid", function (done) {
```

Comment:
nit: since we are testing the new eta should lie after the old eta. can we rename the description to `Should return fail response if the new ETA falls before old ETA`.